### PR TITLE
Fix add_remote_addr and add_http_headers options not working on bulk request

### PR DESCRIPTION
**Pull request by tarokkk**
This small patch solve bulk request parsing. Previously the code tried to insert a string into a list
which caused "no implicit conversion of String into Integer".

### CODE REVIEW

**Comment by Tyrael** [comment]
@repeatedly any chance you could review this PR?

File: lib/fluent/plugin/in_http.rb [hunk 1]
```diff
@@ -131,18 +131,18 @@ def on_request(path_info, params)
...
-          params.each_pair { |k,v|
-            if k.start_with?("HTTP_")
-              record[k] = v
-            end
-          }
-        end
-
-        if @add_remote_addr
-          record['REMOTE_ADDR'] = params['REMOTE_ADDR']
+        if not record.is_a?(Array)
```
Comment (repeatedly) [inline lib/fluent/plugin/in_http.rb]: In fluentd, we use `unless` instead of `if not`.

**Comment by repeatedly** [comment]
Sorry for the delay.

Please add test to `test/plugin/test_in_http.rb`.

**Comment by tarokkk** [comment]
@repeatedly I fixed the code based on your comments and added tests for the headers.

File: lib/fluent/plugin/in_http.rb [hunk 2]
```diff
@@ -131,18 +131,18 @@ def on_request(path_info, params)
...
-            if k.start_with?("HTTP_")
-              record[k] = v
-            end
-          }
-        end
-
-        if @add_remote_addr
-          record['REMOTE_ADDR'] = params['REMOTE_ADDR']
+        unless record.is_a?(Array)
+          if @add_http_headers and
```
Comment (repeatedly) [inline lib/fluent/plugin/in_http.rb]: `and` is needed?

**Comment by repeatedly** [comment]
Commented. Others are OK :+1:

**Comment by repeatedly** [comment]
Thanks for the work! I will release new version soon.

**Comment by repeatedly** [comment]
Released v0.12.16!

**Pull request merged**